File: leetcode/solutions/3754.py
# https://leetcode.com/problems/concatenate-non-zero-digits-and-multiply-by-sum-i/

class Solution:
    def sumAndMultiply(self, n: int) -> int:
        res = 0
        digit_sum = 0

        for digit in str(n):
            digit = ord(digit) - ord("0")

            if digit:
                res = res * 10 + digit
                digit_sum += digit

        return res * digit_sum

File: leetcode/solutions/3754.rs
// https://leetcode.com/problems/concatenate-non-zero-digits-and-multiply-by-sum-i/

impl Solution {
    pub fn sum_and_multiply(n: i32) -> i64 {
        let mut res: i64 = 0;
        let mut digit_sum: i64 = 0;

        for digit in n.to_string().bytes() {
            digit = (digit - 0 as u8) as i64;

            if digit != 0 {
                res = res * 10 + digit;
                digit_sum += digit;
            }
        }

        res * digit_sum
    }
}
